pkg/database/openshiftclusters.go

```diff
@@ -120,6 +124,43 @@ func (c *openShiftClusters) Get(ctx context.Context, key string) (*api.OpenShift
 	}
 }
 
+// QueueLength returns OpenShiftClusters un-queued document count.
+// If error occurs, -1 is returned with error message
+func (c *openShiftClusters) QueueLength(ctx context.Context) (int, error) {
+	partitions, err := c.collc.PartitionKeyRanges(ctx, "OpenShiftClusters")
+	if err != nil {
+		return -1, err
+	}
+
+	var countTotal int
+	for _, r := range partitions.PartitionKeyRanges {
+		// QueryRAW return *map[interface{}]interface{} for sorting
+		result := c.c.QueryWithOptions("", &cosmosdb.Query{
+			Query: `SELECT VALUE COUNT(1) FROM OpenShiftClusters doc WHERE doc.openShiftCluster.properties.provisioningState IN ("Creating", "Deleting", "Updating") AND (doc.leaseExpires ?? 0) < GetCurrentTimestamp() / 1000`,
+		}, &cosmosdb.Options{
+			PartitionID: r.ID,
+		})
+		// because we aggregate count wee don't expect pagination in this query result,
+		// so we gonna call Next() only once per partition.
+		var data struct {
+			Count      int      `json:"_count,omitempty"`
```

Comment:
do you see these fields? with the VALUE keyword can you not unmarshal directly into a `[]string` or `[]int` or something?

Reply:
I think we can drop them at all

Reply:
Our decoder requires fields to be defined. Changed to 
```
	var data struct {
			api.MissingFields
			Document []string `json:"Documents,omitempty"`
		}
```